scripts/zones/Arrapago_Remnants/instances/arrapago_remnants.lua

```diff
@@ -45,67 +45,121 @@ function onInstanceFailure(instance)
-    end
-
-end
-
 function onInstanceComplete(instance)
 end
 
-function onRegionEnter(player,region)
-    if region:GetRegionID() <= 10 then
+function onRegionEnter(player,region,instance)
+    if region:GetRegionID() <= 11 then
         player:startEvent(199 + region:GetRegionID())
     end
 end
 
-function onEventUpdate(entity, eventid, result)
-    if (eventid >= 200 and eventid <= 203) then
-        local instance = entity:getInstance()
-        if instance:getProgress() == 0 then
-            for id = ID.mob[2][eventid-199].mobs_start, ID.mob[2][eventid-199].mobs_end do
-                SpawnMob(id, instance)
-            end
-            instance:setProgress(eventid-199)
+function onInstanceProgressUpdate(instance, progress, elapsed)
+    if instance:getStage() == 1 and progress == 10 then
+        SpawnMob(ID.mob[1][2].rampart, instance)
+    elseif instance:getStage() == 2 and progress == 2 then -- attempt to spawn slot
+        instance:getEntity(bit.band(ID.npc[2][2].SLOT, 0xFFF), dsp.objType.NPC):setStatus(dsp.status.NORMAL)
+    elseif instance:getStage() == 2 and progress == 3 then -- attempt to spawn socket
+        instance:getEntity(bit.band(ID.npc[2][2].SOCKET, 0xFFF), dsp.objType.NPC):setStatus(dsp.status.NORMAL)
+    elseif instance:getStage() == 3 and progress == 1 then
+        SpawnMob(ID.mob[2][0].astrologer, instance)
+    elseif instance:getStage() == 6 and progress == 1 then
+        instance:getEntity(bit.band(ID.npc[6].DOOR, 0xFFF), dsp.objType.NPC):setLocalVar("start",os.time())
+    elseif instance:getStage() == 7 and progress == 0 then
+        local door = instance:getEntity(bit.band(ID.npc[6].DOOR, 0xFFF), dsp.objType.NPC)
+        door:setLocalVar("current",os.time())
+        if (door:getLocalVar("current") - door:getLocalVar("start") <= 420) then
+            SpawnMob(ID.mob[6].treasure_hunter1, instance)
+            SpawnMob(ID.mob[6].treasure_hunter2, instance)
+            SpawnMob(ID.mob[6].qiqirn_mine_1, instance)
+            SpawnMob(ID.mob[6].qiqirn_mine_2, instance)
         end
-    elseif eventid == 204 then
-        -- spawn floor 3
     end
+
 end
 
-function onEventFinish(player, eventid, result)
+function onEventFinish(player, csid, option)
     local instance = player:getInstance()
-    if eventid >= 200 and eventid <= 210 and result == 1 then
-        local ALIGN = player:getAlliance()
-        local POS = player:getPos()
-        if ALIGN ~= nil then
-            for i,v in ipairs(ALIGN) do
+    local chars = instance:getChars()
+    local pos = player:getPos()
+    
+    if csid >= 200 and csid <= 210 and option == 1 then
+        if chars ~= nil then
+            for i,v in ipairs(chars) do
                 if v:getID() ~= player:getID() then
                     v:startEvent(3)
                     v:timer(4000, function(player)
-                    v:setPos(POS.x, POS.y, POS.z, POS.rot)
-                end)
+                    v:setPos(pos.x, pos.y, pos.z, pos.rot)
```

Comment:
Indent this line so that it is easier to skim and see it is part of the timer function above.  I know you did not write this code, but did you try changing the line to:

```lua
    player:setPos(pos.x, pos.y, pos.z, pos.rot)
```

Right now we are in a function spawned by a timer who received a player variable and instead reading the global value to set position.

Reply:
I'll try changing this tonight.

Reply:
`player:setPos(...)` did work, changed it.